### PR TITLE
Feature/rollbar integration

### PR DESCRIPTION
**Pull request by alxtz**
app2

# Issue 
https://github.com/Xfers/Xfers/issues/12059

# Implement
1. add rollbar init script `JavaScript/src/libRollbar.js` (according to rollbar setup
2. add new `SDK_VERSION` injecting variable in `JavaScript/webpack.common.js` & `JavaScript/rollup.config.js`
3. Right now, the error events will only fire on production builds ( `JavaScript/src/initRollbar.js` ), and adding metadata for SDK version (see #Known Issues 1.
4. see the result on https://rollbar.com/Xfers/Xfers-JS-SDK

# Known Issues
1. Since we're only separating production / development builds, any SDK error happens in testing envs like `xfers-fe` , staging , merchant sandbox would all be received on the dashboard
2. The sourcemap hasn't been added (potential issue, the sourcemap from webpack & rollbar are different, also might need to auto update https://github.com/thredup/rollbar-sourcemap-webpack-plugin
3. See if needs additional tags for those events

# Test Cases
1. open SDK on webpack-dev-server with API error, rollbar not updated
2. open SDK on xfers-fe, rollbar receives error with `environment: production`

### CODE REVIEW

File: JavaScript/rollup.config.js [hunk 1]
```diff
@@ -25,7 +27,8 @@ export default {
       ]
     }),
     replace({
-      'process.env.NODE_ENV': JSON.stringify('production')
+      'process.env.NODE_ENV': JSON.stringify('production'),
+      'SDK_VERSION': JSON.stringify(packageJson.version)
```
Comment (rebekahlow-jy) [inline JavaScript/rollup.config.js]: Wow this is nice!

**Comment by alxtz** [comment]
need tracable & reproducable

**Comment by alxtz** [comment]
1. Test filter / tags feature for Sentry / Rollbar
2. Need to be sure if UI & action logging is doable & can be view easily
3. API headers / params are 100% important

**Comment by alxtz** [comment]
Sourcemap is not going to work, since right now we're providing a compiled version of SDK on npm, but when the merchant installs it, and produce their own compiled version of code

The sourcemap we produces is not going to work because our SDK code is already compiled twice, 

https://www.mattzeunert.com/2016/02/14/how-do-source-maps-work.html
https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit#
http://sokra.github.io/source-map-visualization/#custom

Since sourcemap is based on variable text position, our own sourcemap might work when merchant is testing our code in their own dev tool, but it's not going to work in Sentry / Rollbar

**Comment by alxtz** [comment]
# Core objective for this PR

- [ ] Reproducible API error (log API request params / body / url / method, response body in Sentry)
- [ ] Able to replicate UI errors (logging redux actions, current screen, what text of buttons clicked)
- [ ] Add user / environment related tags
- [ ] Filter verbose / not to fix errors 
- [ ] Able to relate issues

**Comment by alxtz** [comment]
image
    Sentry.showReportDialog({ eventId })

**Comment by alxtz** [comment]
Closed in favor for https://github.com/Xfers/xfers-sdk/pull/279

**Pull request closed without merging**